test_extras/test_chemprop/test_neural_fingerprint.py

```diff
@@ -0,0 +1,64 @@
+"""Test Chemprop neural fingerprint."""
+
+import logging
+import unittest
+from typing import Iterable
+
+import torch
+from sklearn.base import clone
+
+from molpipeline.estimators.chemprop.neural_fingerprint import ChempropNeuralFP
+from molpipeline.utils.json_operations import recursive_from_json, recursive_to_json
+
+# pylint: disable=relative-beyond-top-level
+from .chemprop_test_utils.compare_models import compare_params
+from .chemprop_test_utils.constant_vars import NO_IDENTITY_CHECK
+from .chemprop_test_utils.default_models import get_classification_mpnn
+
+logging.getLogger("lightning.pytorch.utilities.rank_zero").setLevel(logging.WARNING)
+
+
+def get_neural_fp_encoder() -> ChempropNeuralFP:
+    """Get the Chemprop model.
+
+    Returns
+    -------
+    ChempropNeuralFP
+        The Chemprop model.
+    """
+    mpnn = get_classification_mpnn()
+    chemprop_model = ChempropNeuralFP(model=mpnn, lightning_trainer__accelerator="cpu")
+    return chemprop_model
+
+
+class TestChempropNeuralFingerprint(unittest.TestCase):
+    """Test the Chemprop model."""
+
+    def test_clone(self) -> None:
+        """Test the clone method."""
+        chemprop_fp_encoder = get_neural_fp_encoder()
+        cloned_encoder = clone(chemprop_fp_encoder)
+        self.assertIsInstance(cloned_encoder, ChempropNeuralFP)
+        compare_params(self, chemprop_fp_encoder, cloned_encoder)
```

Comment:
A good additional test would also be a simple test of a single forward pass and checking that the predictions are np.isclose.

I don't know if this is straightforward to do. Should be only possible if the inference in chemprop is deterministic or can be fixed with a seed. In addition, you don't even need to fit the model for this or with just one sample. Would be good to have this test I think.

Reply:
I seeded torch, numpy and random and still got different results. I dont think it is a straight forwards thing to do.
Plus, clone is not meant to clone a fitted model. If you clone a fitted model, it will still return a unfitted one.